Source/Core/VideoCommon/ShaderGenCommon.h

```diff
@@ -62,6 +59,9 @@ class ShaderGeneratorInterface
 	 */
 	template<class uid_data>
 	uid_data* GetUidData() { return nullptr; }
+
+protected:
+	std::string m_buffer;
```

Comment:
Oh, this is within the superclass? Are you sure std::string() has no overhead? Sounds fine to me, but maybe check if the assembly of the UID generator is still the same.